lib/DBIx/Class/Schema/Loader/Base.pm

```diff
@@ -2626,8 +2631,8 @@ sub _setup_src_meta {
     $self->_dbic_stmt($table_class, 'table', $table->dbic_name);
 
     # Must come after ->table
-    if ($is_view and my $view_def = $self->_view_definition($table)) {
-        $self->_dbic_stmt($table_class, 'result_source_instance->view_definition', $view_def);
+    if (  $is_view and my $view_def = $self->_view_definition( $table ) and not $self->omit_view_definitions ){
```

Comment:
There's no point in getting the view definition if we're not going to emit it, so please move the check before the `->_view_definition` call. Also, please refrain from unrelated whitespace changes.